frontend/src/components/HardwareDialog.tsx

```diff
@@ -1,50 +1,97 @@
 import { useState, useRef, useEffect } from "react";
 
-interface HardwareDialogProps {
-    isOpen: boolean; // Determines if the dialog is open
-    onClose: () => void; // Function to close the dialog
-    onSubmit: (action: "checkOut" | "checkIn", hardwareId: string, quantity: number) => void; // Function to handle submit
+// Define an enum for action types
+enum ActionType {
+    CheckOut = "checkOut",
+    CheckIn = "checkIn",
 }
 
 //Component Setup and State Inits
-const HardwareDialog = ({ isOpen, onClose, onSubmit }: HardwareDialogProps) => {
+export default function HardwareDialog() {
+    const [isOpen, setIsOpen] = useState(false); // State for managing dialog open/close
     const [hardwareId, setHardwareId] = useState(""); // State for hardware ID, or should this projectID (same thing i guess)
     const [quantity, setQuantity] = useState(1); // State for quantity, init to 1
-    const [action, setAction] = useState<"checkOut" | "checkIn">("checkOut"); // Action state
+    const [action, setAction] = useState<ActionType>(ActionType.CheckOut); // Action state
+    const [projectId, setProjectId] = useState(""); // State for project ID
+    //const [authToken, setAuthToken] = useState<string>(""); // State for auth token
+    const dialogRef = useRef<HTMLDialogElement | null>(null);
 
-    const dialogRef = useRef<HTMLDialogElement>(null);
-
-    // Effect to open or close the dialog based on the isOpen prop
+    // Effect to open the dialog when isOpen changes to true
     useEffect(() => {
-        if (isOpen) {
-            //if isOpen is true, than showModal methods opens the dialog
-            dialogRef.current?.showModal();
-        } else {
-            //if isOpen is false, than we close the dialog
-            dialogRef.current?.close();
+        if (isOpen && dialogRef.current) {
+            dialogRef.current.showModal();
+        } else if (!isOpen && dialogRef.current) {
+            dialogRef.current.close();
         }
-    }, [isOpen]); //useEffect triggered whenever isOpen changes
+    }, [isOpen]);
+
+    // Handling form submission
+    const handleSubmit = async (e: React.FormEvent) => {
+        e.preventDefault(); // Prevent default form submission behavior
+        if (!action) return; // Ensure there is an action set
 
-    //Handling form submission
-    const handleSubmit = (e: React.FormEvent) => {
-        e.preventDefault();
-        if (hardwareId && quantity > 0) {
-            onSubmit(action, hardwareId, quantity); // Call onSubmit with action, hardwareId, and quantity
-            setHardwareId(""); // Clear input after submission
-            setQuantity(1); // Reset quantity to default after submission
-            onClose(); // Close dialog after submission
+        //set url depending on check-in or check-out is being called
+        const url = `/api/v1/projects/${projectId}/${hardwareId}/${action === ActionType.CheckOut ? "check-out" : "check-in"}?amount=${quantity}`;
```

Comment:
Other than the previous comment on state, don't worry too much about the actual url for the API. The main thing missing here is the Auth header.

Reply:
Also remember to URL encode things